src/msg/simple/Pipe.cc

```diff
@@ -1342,6 +1344,7 @@ void Pipe::fault(bool onread)
     return;
   }
 
+  recv_reset();
   shutdown_socket();
```

Comment:
is it safe to leave out recv_reset() for the other two callsites of shutdown_socket()? seems like this could use an explanation in its own commit